src/views/DocumentCompare.vue

```diff
@@ -0,0 +1,166 @@
+						{{ documents.textB.name }}
+					</p>
+				</div>
+				<div
+					v-html="highlightedTexts.B"
+					class="px-4 py-1 pt-1 md:pt-2 md:p-4 bg-white text-justify rounded-b-md md:rounded-b-lg shadow-md"
+				/>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script>
+import ResultHeader from '../components/Result/ResultHeader';
+import { escape, cloneDeep } from 'lodash';
+import retry from '@/functions/retry.function';
+
+export default {
+	components: {
+		ResultHeader,
+	},
+	data: function () {
+		return {
+			id: undefined,
+			documentId: undefined,
+			highlightedTexts: undefined,
+			documents: undefined,
+		};
+	},
+	mounted: function () {
+		// initial spinner
+		this.$store.dispatch('setLoading', true);
+
+		this.id = this.$route.params.compare;
+		this.documentId = this.$route.params.document;
+
+		retry(
+			() => this.fetchResult(this.documentId, this.id),
+			(result) => result.textA && result.textB && result.matches
+		)
+			.then((result) => {
+				this.documents = result;
+
+				if (!this.documents.textA || !this.documents.textB)
+					throw new TypeError(
+						'No documents for diff found. Please contact administrator'
+					);
+
+				if (this.documents && this.documents.matches.length === 0)
+					throw new TypeError(
+						'Documents for diff found, but no matches found. Please contact administrator'
+					);
+
+				this.highlightedTexts = this.highlightText(this.documents);
+			})
+			.catch((e) => {
+				this.$store.dispatch('AlertStore/setAlert', {
+					message: e.message,
+					type: 'error',
+					duration: 10000,
+				});
+			})
+			.finally(() => {
+				this.$store.dispatch('setLoading', false);
+			});
+	},
+	methods: {
+		fetchResult(documentId,id) {
+			// Connect to BE
+			return this.$axios
+				.get(`/api/documents/${documentId}/diff/${id}`)
+				.then((response) => response.data);
+
+		},
+		highlightText: function (documents) {
+			const indices = documents.matches
+				.map((matches) => ({
+					fromA: matches.fromA,
+					toA: matches.toA,
+					fromB: matches.fromB,
+					toB: matches.toB,
+					color: this.hex_color_generator(),
+				}))
+				.flat();
+
+			const indicesA = cloneDeep(indices);
+			const indicesB = cloneDeep(indices);
+
+			indicesA.sort((a, b) => b.toA - b.fromA - (a.toA - a.fromA));
+			indicesB.sort((a, b) => b.toB - b.fromB - (a.toB - a.fromB));
+
+			const subStringToReplaceA = indicesA.map((interval) => ({
+				text: documents.textA.content.substring(
+					interval.fromA,
+					interval.toA + 1
+				),
+				color: interval.color,
+			}));
+
+			const subStringToReplaceB = indicesB.map((interval) => ({
+				text: documents.textB.content.substring(
+					interval.fromB,
+					interval.toB + 1
+				),
+				color: interval.color,
+			}));
+
+			return {
+				A: this.merge_text(
+					subStringToReplaceA,
+					documents.textA.content
+				),
+				B: this.merge_text(
+					subStringToReplaceB,
+					documents.textB.content
+				),
+			};
+		},
+		hex_color_generator: function () {
```

Comment:
Have you spoken to @DeMensa123 about the color utility? You're both implementing the same thing.